core-web/libs/block-editor/src/lib/plugins/bubble-link-form.plugin.ts

```diff
@@ -143,81 +167,66 @@ export class BubbleLinkFormView {
         // Get Editor Container Position
         const { element: editorElement } = this.editor.options;
         const editorClientRect = editorElement.parentElement.getBoundingClientRect();
-        const bubbleMenuRect = document.querySelector('#bubble-menu').getBoundingClientRect();
+        const domRect =
+            document.querySelector('#bubble-menu')?.getBoundingClientRect() || nodeClientRect;
 
         // Check for an overflow in the content
         const isOverflow = editorClientRect.bottom < nodeClientRect.bottom;
 
         // Check if the node is a dotImage
         const node = doc?.nodeAt(from);
-        const isNodeImage = node.type.name === 'dotImage';
+        const isNodeImage = node?.type.name === 'dotImage';
 
         // If there is an overflow, use bubble menu position as a reference.
-        return isOverflow || isNodeImage ? bubbleMenuRect : nodeClientRect;
+        return isOverflow || isNodeImage ? domRect : nodeClientRect;
     }
 
-    addLink(link: string) {
-        if (this.isDotImageNode()) {
-            this.editor.commands.setImageLink({ href: link });
-        } else {
-            this.editor.commands.setLink({ href: link });
+    setLinkValues({ link, blank = false }) {
+        if (link.length > 0) {
+            this.isDotImageNode()
+                ? this.editor.commands.setImageLink({ href: link })
+                : this.editor.commands.setLink({ href: link, target: blank ? '_blank' : '_top' });
         }
-        this.hide();
     }
 
     removeLink() {
-        if (this.isDotImageNode()) {
-            this.editor.commands.unsetImageLink();
-        } else {
-            this.editor.commands.unsetLink();
-        }
+        this.isDotImageNode()
+            ? this.editor.commands.unsetImageLink()
+            : this.editor.commands.unsetLink();
         this.hide();
     }
 
     setInputValues() {
-        this.component.instance.nodeLink = this.getNodeLink();
-        this.component.instance.newLink = this.getNodeLink() || this.getLinkSelect();
-    }
-
-    focusInput() {
-        this.component.instance.focusInput();
+        const values = this.getLinkProps();
+        this.component.instance.initialValues = values;
+        this.component.instance.setFormValue(values);
     }
 
     setComponentEvents() {
-        this.component.instance.hideForm.subscribe(() => this.hide());
+        this.component.instance.hide.subscribe(() => this.hide());
         this.component.instance.removeLink.subscribe(() => this.removeLink());
-        this.component.instance.setLink.subscribe((event) => this.addLink(event));
+        this.component.instance.setNodeProps.subscribe((event) => this.setLinkValues(event));
```

Comment:
Do we need a `takeUntil` here? this looks like something that is going to susbscribe everytime open?